OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -220,7 +231,7 @@ ActorInfo ChooseBuildingToBuild(ProductionQueue queue)
 			// Next is to build up a strong economy
 			if (!baseBuilder.HasAdequateRefineryCount)
 			{
-				var refinery = GetProducibleBuilding(baseBuilder.Info.RefineryTypes, buildableThings);
+				var refinery = buildableConsiderations.FirstOrDefault(c => baseBuilder.Info.RefineryTypes.Contains(c.Name));
 				if (refinery != null && HasSufficientPowerForActor(refinery))
 				{
 					AIUtils.BotDebug("AI: {0} decided to build {1}: Priority override (refinery)", queue.Actor.Owner, refinery.Name);
```

Comment:
The automated considerations (power and economy management) somewhat conflict with explicit considerations (I do understand that both can be combined). I'm mentioning it because currently, when you explicitly let the AI build a refinery in the early game (`baseBuilder.HasAdequateRefineryCount == false`), the debug message says the decision would have been a priority override. 

IMO it would make sense to support disabling the automated considerations on the `BaseBuilderBotModule`, so that you could have an instance with automated management for later phases of the game and an instance without automated management for early phases.